regression/strings-smoke-tests/java_format6/test.java

```diff
@@ -0,0 +1,127 @@
+public class test {
+    public static void check_boolean() {
+        String s = String.format("foo %b", true);
+        assert(s.equals("foo true"));
```

Comment:
Nitpicking: The brackets are not necessary because `assert` is not considered a function in Java.

You could enhance that test by adding the following line:
`assert !s.equals("foo true");`
That would check that the string solver doesn't over-constrain. However, checking the negation of the assertion is generally longer in cbmc, so you first have to check that the test doesn't take too long (< 3 sec).